doc/whats_new/v0.22.rst

```diff
@@ -14,17 +14,56 @@ refer to
 :ref:`sphx_glr_auto_examples_release_highlights_plot_release_highlights_0_22_0.py`.
 
 
+Clear definition of the public API
+----------------------------------
+
+Scikit-learn has a public API, and a private API.
+
+We do our best not to break the public API, and to only introduce
+backward-compatible changes that do not require any user action. However, in
+cases where that's not possible, any change to the public API is subject to
+a deprecation cycle of two minor versions. The private API isn't publicly
+documented and isn't subject to any deprecation cycle, so users should not
+rely on its stability.
+
+Whether a tool is private or public depends on whether you can import it
+without a leading underscore in the import path. For example
+``sklearn.pipeline.make_pipeline`` is public, while
+`sklearn.pipeline._name_estimators` is private.
+``sklearn.ensemble._gb.BaseEnsemble`` is private too because the whole `_gb`
+module is private.
+
+Up to 0.22, some tools were de-facto public (no leading underscore), while
+they should have been private in the first place. In version 0.22, these
+tools have been made properly private, and the public API space has been
+cleaned. In addition, importing from files in sub-packages is deprecated:
+you should use ``from sklearn.cluster import Birch`` instead of ``from
```

Comment:
This can be confusing because of e.g. `sklearn.feature_extraction.text` or `sklearn.metrics.pairwise` that are not going anyway. Whether the module is a file or a folder doesn't matter to Python. Maybe say instead that,
```
In addition, some sub-modules were deprecated: you should use ...
``